scripts/extract_pdf_with_pdftotext.sh

```diff
@@ -3,7 +3,35 @@
 # Call the PDF extractor and returns (writes into STDOUT) its output
 
 if [ ! -e $1 ]; then
-	exit 127 # not found
+        exit 127 # not found
 fi
 
-/usr/bin/pdftotext -enc UTF-8 -eol unix -q "$1"  -
+EXITCODE=0
+STD_OUT="$(mktemp)"
+
+#/usr/bin/pdftotext -enc UTF-8 -eol unix -q "$1"  -
+
+timeout -s SIGINT 30s /usr/bin/pdftotext -enc UTF-8 -eol unix -q "$1"  - > $STD_OUT
```

Comment:
I would make the timeout configurable via env variables so that we can fine tune the value if/when needed. For example, we could define at the beginning of the script this variable with a default value:

```
PDFTOTEXT_TIMEOUT="${PDFTOTEXT_TIMEOUT:30s}"
```

And then use it every time we call `timeout` as `${PDFTOTEXT_TIMEOUT}`.

Reply:
Introduced timeout environmental variable, added some comments.